### ISSUE TITLE
paranamer with gradle

### ISSUE THREAD

**Issue by rils** [paul-hammant/paranamer#30]
I tried to integrate in my Gradle based Android application project. Seems not working.
Added apranamer-ant,paranamer-generator jar as external libraries and added below in app module build.gradle file. But seems not working. Could someone give a proper example ?

 `    configurations {
        paranamer
    } 
tasks.withType(JavaCompile) {
    it.doLast {
        ant.taskdef name: 'paranamer',
                classname: 'com.thoughtworks.paranamer.ant.ParanamerTask',
                classpath: configurations.paranamer.asPath
        ant.paranamer(classdir: destinationDir) {
            source.addToAntBuilder ant, 'fileset', FileCollection.AntType.FileSet
        }
    }
}`

**Comment by paul-hammant** [comment]
I'll give it a go, but help me help you.

Can you make a "hello-world" version of a Gradle-build app, that fails or doesn't work in the same way as your problem shows ..... and I'll do my best to advise you on fixing it, or fixing Paranamer

**Comment by rils** [comment]
Hello Paul,
Thanks!! I figured out why it was not working. I made a HelloWorld app for you to debug by applying all gradle settings. Surprisingly paranmer worked.  I went back to my app and checked the differences. I had enabled `jack` to make my app compiler to JAVA8. HelloWorld app was default JAVA which was not JAVA8. I removed JAVA8 option from my app and it worked liek charm :) . Thanks!
I pushed the helloworld app to my github with necessary gradle changes to demonstrate paranamer integration into gradle and android. May be it will be helpful for other users. JAVA8 will work only from Android O. so the apps before android O still can use paranemer to fetch parameter data.
[Paranamer-Android-Gradle](https://github.com/rils/Paranamer-Android-Gradle)
